app-drawer/app-drawer.html

```diff
@@ -311,9 +336,22 @@
         this._setPosition(this.align);
       },
 
+      _focusHandler: function(event) {
+        if (this._isAboveContent() && !this.contains(Polymer.dom(event).rootTarget)) {
```

Comment:
`contains` doesn't cross ShadowDOM boundaries.  There is [deepContains](https://github.com/Polymer/polymer/blob/master/src/lib/dom-api.html#L41) that ignores ShadowDOM boundaries.  But that also will not work if the focused elements are being distributed into `app-drawer`, e.g.
```
<app-drawer>
  <content></content>
</app-drawer>
```

Using event path is probably the only option here.